pkg/kv/kvserver/kvflowcontrol/rac2/range_controller.go

```diff
@@ -2306,7 +2306,8 @@ func (rs *replicaState) createReplicaSendStreamRaftMuLocked(
 	rs.parent.opts.ReplicaMutexAsserter.RaftMuAssertHeld()
 	// Must be in StateReplicate on creation.
 	if log.ExpensiveLogEnabled(ctx, 1) {
-		log.VEventf(ctx, 1, "creating send stream %v for replica %v", rs.stream, rs.desc)
+		log.VEventf(ctx, 1, "r%v creating send stream %v for replica %v",
```

Comment:
Did you get to the bottom of why `ctx` doesn't have the range ID? Is's the [raftCtx](https://github.com/cockroachdb/cockroach/blob/661e7c7fb93b7ffb362cff32f27670012a17492c/pkg/kv/kvserver/replica_init.go#L196), right? Wonder why we don't annotate it.

Reply:
Perhaps there are some code paths that call with with a different context. I found [this](https://github.com/cockroachdb/cockroach/blob/2b5a6f7b036bfda258f4f31f008f56b27e8243b0/pkg/kv/kvserver/store_snapshot.go#L1707-L1712) stack, for example.

Reply:
I had similar thinking, that even if the `raftCtx` was reliably annotated with the `range_id` we'd still have a problem where some log lines could be hit without using the `raftCtx`.